src/providers/ldap/ldap_id_subid.c

```diff
@@ -0,0 +1,264 @@
+    if (!req) return NULL;
+
+    state->ev = ev;
+    state->ctx = ctx;
+    state->sdom = sdom;
+    state->conn = conn;
+    state->dp_error = DP_ERR_FATAL;
+    state->name = talloc_strdup(state, filter_value);
+    if (!state->name) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_strdup failed\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    state->op = sdap_id_op_create(state, state->conn->conn_cache);
+    if (!state->op) {
+        DEBUG(SSSDBG_OP_FAILURE, "sdap_id_op_create failed\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    state->domain = sdom->dom;
+
+    state->filter = talloc_asprintf(state,
+                                    "(&(%s=%s)(%s=%s))",
+                                    SYSDB_OBJECTCLASS,
+                                    ctx->opts->subid_map[SDAP_OC_SUBID_RANGE].name,
+                                    ctx->opts->subid_map[SDAP_AT_SUBID_RANGE_OWNER].name,
+                                    extra_value);
+
+    ret = subid_ranges_get_retry(req);
+    if (ret != EOK) {
+        goto done;
+    }
+
+    return req;
+
+done:
+    if (ret != EOK) {
+        tevent_req_error(req, ret);
+    } else {
+        tevent_req_done(req);
+    }
+    return tevent_req_post(req, ev);
+}
+
+
+static int subid_ranges_get_retry(struct tevent_req *req)
+{
+    struct subid_ranges_get_state *state = tevent_req_data(req,
+                                                    struct subid_ranges_get_state);
+    struct tevent_req *subreq;
+    int ret = EOK;
+
+    subreq = sdap_id_op_connect_send(state->op, state, &ret);
+    if (!subreq) {
+        return ret;
+    }
+
+    tevent_req_set_callback(subreq, subid_ranges_get_connect_done, req);
+    return EOK;
+}
+
+
+static void subid_ranges_get_connect_done(struct tevent_req *subreq)
+{
+    struct tevent_req *req = tevent_req_callback_data(subreq,
+                                                      struct tevent_req);
+    struct subid_ranges_get_state *state = tevent_req_data(req,
+                                                     struct subid_ranges_get_state);
+    int dp_error = DP_ERR_FATAL;
+    int ret;
+
+    ret = sdap_id_op_connect_recv(subreq, &dp_error);
+    talloc_zfree(subreq);
+
+    if (ret != EOK) {
+        state->dp_error = dp_error;
+        tevent_req_error(req, ret);
+        return;
+    }
+
+    subid_ranges_get_search(req);
+}
+
+
+static void subid_ranges_get_search(struct tevent_req *req)
+{
+    struct subid_ranges_get_state *state = tevent_req_data(req,
+                                                     struct subid_ranges_get_state);
+    struct tevent_req *subreq = NULL;
+    const char **attrs;
+    int ret;
+
+    ret = build_attrs_from_map(state, state->ctx->opts->subid_map,
+                               SDAP_OPTS_SUBID_RANGE, NULL, &attrs, NULL);
+    if (ret != EOK) {
+        tevent_req_error(req, ENOMEM);
+        return;
+    }
+
+    subreq = sdap_get_generic_send(state,
```

Comment:
More then one search base can be defined, you need to iterate over all. Use `sdap_search_bases()`?

Reply:
If I understand correctly, design of current solution in FreeIPA doesn't support multiple bases, so supporting this in SSSD doesn't make a lot of sense.

But formally speaking you are correct, of course. I just was lost in sdap code and tried to implement minimal operable solution (as I wasn't sure approach is viable at all).

I'll try to expand this now to support multiple bases.

Reply:
In particular, I was looking at "deskprofile" that seems to expect single search base specified:
https://github.com/SSSD/sssd/blob/7fdff741a7acaf016390988e5795df6bf03e7605/src/providers/ipa/ipa_deskprofile_config.c#L64

Reply:
I took `sdap_async_sudo` as an example and added one patch to make use of `sdap_search_bases`.
(Patch to be squashed later.)